### PR TITLE
Typescript port connection, component, connectionTarget

### PR DESCRIPTION
**Pull request by jackruan99**

### CODE REVIEW

File: src/app/core/connection.ts [hunk 1]
```diff
@@ -671,7 +666,7 @@ export default class Connection {
      * @memberof Connection
      * @returns {void}
      */
-    setSourceFromJSON(device: Device, json: JSON) {
+    setSourceFromJSON(device: Device, json: InterchangeV1) {
```
Comment (rkrishnasanka) [inline src/app/core/connection.ts:669]: This should be `ConnectionTargetInterchangeV1`

File: src/app/core/component.ts [hunk 2]
```diff
@@ -172,12 +171,23 @@ export default class Component {
...
-        output.layers = this.__findLayerReferences();
+        const output: ConnectionInterchangeV1 = {
+            id : this._id,
+            name: this._name,
+            entity: this._entity,
+            source: null,
+            sinks: null,
+            params: this._params.toJSON(),
+            xspan: bounds.width,
+            yspan: bounds.height,
```
Comment (rkrishnasanka) [inline src/app/core/component.ts:182]: Dont pull from bounds, get from xspan, yspan

File: src/app/core/component.ts [hunk 3]
```diff
@@ -240,7 +250,7 @@ export default class Component {
      * @memberof Component
      */
     getPosition() {
```
Comment (rkrishnasanka) [inline src/app/core/component.ts:252]: Add return type

File: src/app/core/component.ts [hunk 4]
```diff
@@ -265,31 +275,28 @@ export default class Component {
...
+    // /**
+    //  * Checks if the component has default parameters
+    //  * @param {String} key Key to access the component
+    //  * @returns {boolean} Returns true whether it has default parameters or not
+    //  * @memberof Component
+    //  */
+    // hasDefaultParam(key: string) {
+    //     if (this.getDefaults().hasOwnProperty(key)) return true;
+    //     else return false;
+    // }
```
Comment (rkrishnasanka) [inline src/app/core/component.ts:290]: if its not getting called, delete it

File: src/app/core/component.ts [hunk 5]
```diff
@@ -451,7 +456,7 @@ export default class Component {
         this.xspan = this._xspan;
         this.yspan = this._yspan;
 
-        const params = {};
+        const params = json.params
         if (entity === "TEST MINT") {
             console.warn("Found legacy invalid entity string", entity);
```
Comment (rkrishnasanka) [inline src/app/core/component.ts:461]: you can delete this

File: src/app/core/component.ts (hunk 4 above)
Comment (jackruan99) [reply src/app/core/component.ts:290]: I didn't realize that it is actually being used

File: src/app/core/component.ts [hunk 6]
```diff
@@ -522,8 +526,8 @@ export default class Component {
      * @memberof Component
      * @returns {void}
      */
-    setPort(label, port) {
-        this.__ports.set(label, port);
+    setPort(label: string, port: Port) {
```
Comment (rkrishnasanka) [inline src/app/core/component.ts:529]: This should be type `ComponentPort` not `Port`

File: src/app/core/registry.ts [hunk 7]
```diff
@@ -1,16 +1,21 @@
...
+import FeatureSet from "../featureSets/featureSet";
 
 class Registry {
     /*
     Place where we store the data necessary for the text label
     TODO: Change this from this awful hacky implementation
   */
+  constructor(){
+    registerSets()
+  }
```
Comment (rkrishnasanka) [inline src/app/core/registry.ts:12]: Is this solving your problems with `FeatureSet` ?

**Review by rkrishnasanka — COMMENTED** [review]
@murrayLF Are the `params.ts` changes here okay ?

File: src/app/core/connectionTarget.ts [hunk 8]
```diff
@@ -40,13 +43,13 @@ export default class ConnectionTarget {
...
             return {
-                component: this.__component.getID(),
-                port: this.__portLabel
+                component: this._component.id,
+                port: this._portLabel
             };
         } else {
-            return { component: this.__component, port: this.__portLabel };
+            return { component: this._component, port: this._portLabel };
         }
```
Comment (rkrishnasanka) [inline src/app/core/connectionTarget.ts:53]: We are no longer supporting older design data right? so only the first option should be available

File: src/app/core/component.ts [hunk 9]
```diff
@@ -532,9 +536,9 @@ export default class Component {
...
      */
     getRotation() {
-        if (this.__params.hasParam("rotation")) {
+        if (this._params.hasParam("rotation")) {
             return this.getValue("rotation");
-        } else if (this.__params.hasParam("orientation")) {
+        } else if (this._params.hasParam("orientation")) {
             const orientation = this.getValue("orientation");
             if (orientation === "V") {
                 return 0;
```
Comment (rkrishnasanka) [inline src/app/core/component.ts:544]: Let get rid of this orientation parameter check. this is once again for legacy components

File: src/app/core/component.ts [hunk 10]
```diff
@@ -451,13 +456,12 @@ export default class Component {
         this.xspan = this._xspan;
         this.yspan = this._yspan;
 
-        const params = {};
-        if (entity === "TEST MINT") {
-            console.warn("Found legacy invalid entity string", entity);
-            entity = name.split("_")[0]; // '^.*?(?=_)'
+        const params = json.params
+
+        entity = name.split("_")[0]; // '^.*?(?=_)'
```
Comment (rkrishnasanka) [inline src/app/core/component.ts:461]: you shouldn't have to do split or anything, entity should just be taken from the entity property of the json

**Comment by rkrishnasanka** [comment]
@jackruan99 @MurrayLF can I close this PR ? I'm guessing all the changes here are going to come in with #296

**Comment by rkrishnasanka** [comment]
Merged with the other typescript integration PR

**Pull request closed without merging**